diff --git a/mailchimp.module b/mailchimp.module
index ebc7867..0f3bdda 100644
--- a/mailchimp.module
+++ b/mailchimp.module
@@ -308,11 +308,19 @@ function mailchimp_get_memberinfo($list_id, $email, $reset = FALSE) {
     }
   }
   catch (Exception $e) {
-    \Drupal::logger('mailchimp')->error('An error occurred requesting memberinfo for {email} in list {list}. "{message}"', array(
-      '@email' => $email,
-      '@list' => $list_id,
-      '%message' => $e->getMessage(),
-    ));
+    // A 404 exception code means mailchimp does not have subscription
+    // information for given email address. This is not an error and we can
+    // cache this information.
+    if ($e->getCode() == 404) {
+      $cache->set($list_id . '-' . $email, $memberinfo);
+    }
+    else {
+      \Drupal::logger('mailchimp')->error('An error occurred requesting memberinfo for {email} in list {list}. "{message}"', array(
+        '@email' => $email,
+        '@list' => $list_id,
+        '%message' => $e->getMessage(),
+      ));
+    }
   }
 
   return $memberinfo;
